Use parenthesis instead of curly braces for Resource ctor.
I don't think the original code is invalid C++, but nvcc is getting confused and errors out with
strange parsing issues.

author: micketeer@gmail.com
--- tests/integration/resource_aware_pool_tests.cpp.orig	2026-10-13 02:25:18.111296859 +0200
+++ tests/integration/resource_aware_pool_tests.cpp	2026-10-13 02:25:39.360974740 +0200
@@ -358,12 +358,12 @@
   
   // Now it's free - should return warning and default
   camp::resources::Resource res = get_resource(m_pool, ptr);
-  EXPECT_TRUE(res == camp::resources::Resource{Host{}});
+  EXPECT_TRUE(res == camp::resources::Resource(Host{}));
   
   // Invalid pointer
   double* invalid_ptr = reinterpret_cast<double*>(0xDEADBEEF);
   res = get_resource(m_pool, invalid_ptr);
-  EXPECT_EQ(res, camp::resources::Resource{Host{}});
+  EXPECT_EQ(res, camp::resources::Resource(Host{}));
 }
 
 TEST_P(ResourceAwarePoolTest, StressTestMultipleResources)
